scanner/src/lib/dataproducts/MobileWorkMachine_Drill_ManufacturingDataSheet_v0.2.svelte

```diff
@@ -0,0 +1,173 @@
+<script lang="ts">
+  /*
+      Docs:
+
+      https://github.com/ioxio-dataspace/sandbox-definitions/blob/main/src/DigitalProductPassport/MobileWorkMachine/Drill/ManufacturingDataSheet_v0.2.py
+
+      https://definitions.sandbox.ioxio-dataspace.com/definitions/DigitalProductPassport/MobileWorkMachine/Drill/ManufacturingDataSheet_v0.2
+
+      https://gateway.sandbox.ioxio-dataspace.com/docs#/Data%20Products/DigitalProductPassport_MobileWorkMachine_Drill_ManufacturingDataSheet_v0_2_DigitalProductPassport_MobileWorkMachine_Drill_ManufacturingDataSheet_v0_2_post
+       */
+
+  import { countryListAlpha3, formatNumber } from "$lib/common"
+  import DataRow from "$components/DataRow/index.svelte"
+  import Electric from "$assets/fully-electric.svg"
+  import Hybrid from "$assets/hybrid.svg"
+  import FuelPowered from "$assets/fuel-powered.svg"
+
+  enum PowerSystemType {
+    FULLY_ELECTRIC = "fully electric",
+    HYBRID = "hybrid",
+    FUEL_POWERED = "fuel powered",
+  }
+
+  export let status: number
+  export let data: {
+    productName: string
+    boomCoverage: number
+    trammingDistance: number
+    maximumHoleLength: number
+    minimumHoleDiameter: number
+    maximumHoleDiameter: number
+    drillingPower: number
+    referenceDataSheet: string
+    safetyDataSheet: string
+    manufacturerInformation: {
+      name: string
+      streetName: string
+      postalCode: string
+      city: string
+      country: string
+      website: string
+      email: string
+    }
+    powerSystem: {
+      type: PowerSystemType
+      electricMotors: {
+        motorType: string
+        count: number
+      }[]
+      batteries: {
+        power: number
+        cellType: string
+        count: number
+      }[]
+    }
+  }
+  let powerSystemSvg: string | null
+  switch (data.powerSystem.type) {
+    case PowerSystemType.FULLY_ELECTRIC:
+      powerSystemSvg = Electric
+      break
+    case PowerSystemType.HYBRID:
+      powerSystemSvg = Hybrid
+      break
+    case PowerSystemType.FUEL_POWERED:
+      powerSystemSvg = FuelPowered
+      break
+    default:
+      powerSystemSvg = null
+  }
```

Comment:
Let's not use switch-case for something like this, it takes twice the lines of code than
```
let powerSystemSvg = null
if (data.powerSystem.type === PowerSystemType.FULLY_ELECTRIC) {
  powerSystemSvg = Electric
} ...
```

Reply:
honestly there are very few places I would want to see a switch-case in nowadays

Reply:
hell, just use a map

Reply:
```
const powerSystemIcon = {
  PowerSystemType.FULLY_ELECTRIC: Electric,
  ...
}
const powerSystemSvg = powerSystemIcon[data.powerSystem.type] || null
```